System/Random.hs

```diff
@@ -859,10 +862,10 @@ applyAtomicGen op (AtomicGenI gVar) =
 -- not being used with `IOGen`. Which also means that it is not safe in a
 -- concurrent setting.
 --
--- Both `IOGen` and `AtomicGen` are necessary when generation of random values
--- happens in `IO` and especially when dealing with exception handling and
--- resource allocation, which is where `StateT` should never be used. For
--- example writing a random number of bytes into a temporary file:
+-- Both `IOGen` and `AtomicGen` are necessary when generation of pseudo-random
+-- values happens in `IO` and especially when dealing with exception handling
+-- and resource allocation, which is where `StateT` should never be used. For
+-- example writing a pseudo-random number of bytes into a temporary file:
```

Comment:
@curiousleo you see this paragraph mentions that both `IOGen` and `AtomicGen` are to be used when dealing with exceptions

Reply:
When reading this, I stumbled over "Both `IOGen` and `AtomicGen` are necessary" because you would only ever use one of them, not both together, right? Then I gave up on trying to understand the rest of it. The part about "don't use `StateT` when there are exceptions" is clear though, thanks for pointing me here.

Reply:
That's right. I mean, you could use them together at the same time to track two separate RNGs, but there is no requirement to use them together at the same time. Maybe saying `"Generators like `IOGen` and `AtomicGen` that work in `IO` are necessary when dealing with exceptions and resource allocation ..."`

Reply:
I'll tackle this in a follow-up PR.